tests/PHPStan/Analyser/nsrt/uppercase-string-parse.php

```diff
@@ -18,7 +18,7 @@ public function parse(string $uppercase, string $string): void
 		if (array_key_exists('foo', $a)) {
 			$value = $a['foo'];
 			if (\is_string($value)) {
-				assertType('uppercase-string', $value);
+				assertType('lowercase-string|uppercase-string', $value);
```

Comment:
This is wrong. I assume it's because of the dynamicReturnType https://github.com/phpstan/phpstan-src/pull/3613#discussion_r1835753441

Can you try
```
@param-out (
$string is lowercase-string&uppercase-string 
? array<int|string, array<mixed>|lowercase-string&uppercase-string>
: ($string is lowercase-string
    ? array<int|string, array<mixed>|lowercase-string>
    : ($string is uppercase-string
        ? array<int|string, array<mixed>|uppercase-string>
        : array<int|string, array<mixed>|string>
        )
     )
) $result
```

Reply:
Same strategy for
-  https://github.com/phpstan/phpstan-src/pull/3613#discussion_r1835753838
- https://github.com/phpstan/phpstan-src/pull/3613#discussion_r1835753877
- https://github.com/phpstan/phpstan-src/pull/3613#discussion_r1835753900

Might solve the trim issue.

But the change in those test need to be reverted.

Reply:
No problem, this is not an easy PR.
The cohabitation of both lowercase-string and uppercase-string is harder than introducing only one of those accessory types.

Reply:
Tests reverted in e99bf4f85